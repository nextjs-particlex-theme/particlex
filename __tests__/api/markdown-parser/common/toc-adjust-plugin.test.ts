import adjustToc from '@/api/markdown-parser/common/toc-adjust-plugin'
import rehypeStringify from 'rehype-stringify'
import remarkGfm from 'remark-gfm'
import remarkParse from 'remark-parse'
import remarkRehype from 'remark-rehype'
import { unified } from 'unified'


test('Toc adjust', async () => {
  const doc = `# h1
  
ee

## h2

eee
  `
  const parsed = await unified()
    .use(remarkParse)
    .use(remarkGfm)
    .use(adjustToc)
    .use(remarkRehype)
    .use(rehypeStringify)
    .process(doc)
  expect(String(parsed)).toMatchSnapshot()
})

test('No toc adjust', async () => {
  const doc = `## h1
  
ee

### h2

eee
  `
  const parsed = await unified()
    .use(remarkParse)
    .use(remarkGfm)
    .use(adjustToc)
    .use(remarkRehype)
    .use(rehypeStringify)
    .process(doc)
  expect(String(parsed)).toMatchSnapshot()
})