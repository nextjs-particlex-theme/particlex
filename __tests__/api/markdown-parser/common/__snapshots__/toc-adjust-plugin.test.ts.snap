// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`No toc adjust 1`] = `
"<h2>h1</h2>
<p>ee</p>
<h3>h2</h3>
<p>eee</p>"
`;

exports[`Toc adjust 1`] = `
"<h2>h1</h2>
<p>ee</p>
<h3>h2</h3>
<p>eee</p>"
`;
